examples/gitserver/README.md

```diff
@@ -1,16 +1,220 @@
 Configurable Git Server
 =======================
 
-This example provides automatic mirroring of Git repositories, intended
-for use within a container or Kubernetes pod. It can clone repositories
-from remote systems on startup as well as remotely register hooks. It
-can automatically initialize and receive Git directories on push.
-
-In the more advanced modes, it can integrate with an OpenShift server to
-automatically perform actions when new repositories are created, like
-reading the build configs in the current namespace and performing
-automatic mirroring of their input, and creating new build-configs when
-content is pushed.
-
-The Dockerfile built by this example is published as
-openshift/origin-gitserver
\ No newline at end of file
+Overview
+--------
+
+This example git server is intended for use within a container or Kubernetes pod.
+It can clone repositories from remote systems on startup as well as automatically
+launch builds in an OpenShift project. It can automatically initialize and receive
+Git directories on push.
+
+Hooks can be customized to perform actions on push such as invoking `oc new-app` on a
+repository's source.
+
+The Dockerfile built by this example is published as openshift/origin-gitserver
+
+Quick Start
+-----------
+
+Prerequisites:
+
+* You have an OpenShift v3 server running
+* You are logged in and have access to a project
+* You have the `gitserver.yaml` in this directory
+* You can create externally accessible routes on your server
+
+**NOTE:** It is strongly recommended to use a secure route for your server. Otherwise
+all code and credentials are passed in plain text between your client and the server.
+
+### Deploy the Git Server
+
+1. Create the Git Server
+
+    ```sh
+    $ oc create -f gitserver.yaml
+    ```
+
+2. Grant `view` access to the `gitserver` service account
```

Comment:
I think we should call it the `git` service account and the service `git` and the route `git`.  And probably the dc and a bunch of other things.